test/Graphics/VertexBuffer.test.cpp

```diff
@@ -8,3 +15,162 @@ static_assert(std::is_move_constructible_v<sf::VertexBuffer>);
+            const sf::VertexBuffer vertexBuffer(sf::VertexBuffer::Static);
+            CHECK(vertexBuffer.getVertexCount() == 0);
+            CHECK(vertexBuffer.getNativeHandle() == 0);
+            CHECK(vertexBuffer.getPrimitiveType() == sf::PrimitiveType::Points);
+            CHECK(vertexBuffer.getUsage() == sf::VertexBuffer::Static);
+        }
+
+        SUBCASE("Primitive type and usage constructor")
+        {
+            const sf::VertexBuffer vertexBuffer(sf::PrimitiveType::LineStrip, sf::VertexBuffer::Dynamic);
+            CHECK(vertexBuffer.getVertexCount() == 0);
+            CHECK(vertexBuffer.getNativeHandle() == 0);
+            CHECK(vertexBuffer.getPrimitiveType() == sf::PrimitiveType::LineStrip);
+            CHECK(vertexBuffer.getUsage() == sf::VertexBuffer::Dynamic);
+        }
+    }
+
+    SUBCASE("Copy semantics")
+    {
+        const sf::VertexBuffer vertexBuffer(sf::PrimitiveType::LineStrip, sf::VertexBuffer::Dynamic);
+
+        SUBCASE("Construction")
+        {
+            const sf::VertexBuffer vertexBufferCopy(vertexBuffer);
+            CHECK(vertexBufferCopy.getVertexCount() == 0);
+            CHECK(vertexBufferCopy.getNativeHandle() == 0);
+            CHECK(vertexBufferCopy.getPrimitiveType() == sf::PrimitiveType::LineStrip);
+            CHECK(vertexBufferCopy.getUsage() == sf::VertexBuffer::Dynamic);
+        }
+
+        SUBCASE("Assignment")
+        {
+            sf::VertexBuffer vertexBufferCopy;
+            vertexBufferCopy = vertexBuffer;
+            CHECK(vertexBufferCopy.getVertexCount() == 0);
+            CHECK(vertexBufferCopy.getNativeHandle() == 0);
+            CHECK(vertexBufferCopy.getPrimitiveType() == sf::PrimitiveType::LineStrip);
+            CHECK(vertexBufferCopy.getUsage() == sf::VertexBuffer::Dynamic);
+        }
+    }
+
+    SUBCASE("create()")
+    {
+        sf::VertexBuffer vertexBuffer;
+        CHECK(vertexBuffer.create(100));
+        CHECK(vertexBuffer.getVertexCount() == 100);
+    }
+
+    SUBCASE("update()")
+    {
+        sf::VertexBuffer            vertexBuffer;
+        std::array<sf::Vertex, 128> vertices{};
+
+        SUBCASE("Vertices")
+        {
+            SUBCASE("Uninitialized buffer")
+            {
+                CHECK(!vertexBuffer.update(vertices.data()));
+            }
+
+            CHECK(vertexBuffer.create(128));
+
+            SUBCASE("Null vertices")
+            {
+                CHECK(!vertexBuffer.update(nullptr));
+            }
+
+            CHECK(vertexBuffer.update(vertices.data()));
+            CHECK(vertexBuffer.getVertexCount() == 128);
+            CHECK(vertexBuffer.getNativeHandle() != 0);
+        }
+
+        SUBCASE("Vertices, count, and offset")
+        {
+            CHECK(vertexBuffer.create(128));
+
+            SUBCASE("Count + offset too large")
+            {
+                CHECK(!vertexBuffer.update(vertices.data(), 100, 100));
+            }
+
+            CHECK(vertexBuffer.update(vertices.data(), 128, 0));
```

Comment:
Would it make sense to test something other than 0 offset, to see that the param actually has an effect?

I guess this might need some test data that isn't just default initialized vertices.

Reply:
How would we observe the effect of an offset without drawing the buffer which is out of scope of the tests at the moment?